File: ui/article-aem-renderer/src/test/java/org/cru/godtools/article/aem/db/BaseArticleRoomDatabaseIT.java
package org.cru.godtools.article.aem.db;

import org.junit.After;
import org.junit.Before;

import androidx.room.Room;
import androidx.test.core.app.ApplicationProvider;

public abstract class BaseArticleRoomDatabaseIT {
    ArticleRoomDatabase mDb;

    @Before
    public final void initDb() {
        mDb = Room.inMemoryDatabaseBuilder(ApplicationProvider.getApplicationContext(), ArticleRoomDatabase.class)
                .allowMainThreadQueries()
                .build();
    }

    @After
    public final void cleanupDb() {
        mDb.close();
    }
}

File: ui/article-aem-renderer/src/test/java/org/cru/godtools/article/aem/db/BaseArticleRoomDatabaseIT.kt
package org.cru.godtools.article.aem.db

import androidx.room.Room
import androidx.test.core.app.ApplicationProvider
import org.junit.After
import org.junit.Before

abstract class BaseArticleRoomDatabaseIT {
    protected lateinit var db: ArticleRoomDatabase

    @Before
    fun createDb() {
        db = Room.inMemoryDatabaseBuilder(ApplicationProvider.getApplicationContext(), ArticleRoomDatabase::class.java)
            .allowMainThreadQueries()
            .build()
    }

    @After
    fun closeDb() = db.close()
}
